### PR TITLE
Use single Couchbase bucket

### PR DESCRIPTION
**Pull request by dmytro-lazebnyi**

### CODE REVIEW

File: scalecube-organization-api/src/main/java/io/scalecube/account/api/OrganizationMember.java [hunk 1]
```diff
@@ -45,6 +46,9 @@ public int hashCode() {
 
   @Override
   public String toString() {
-    return super.toString() + String.format("[userId=%s, role=%s]", id, role);
+    return new StringJoiner(", ", OrganizationMember.class.getSimpleName() + "[", "]")
+        .add("id='" + id + "'")
```
Comment (aharonha) [inline scalecube-organization-api/src/main/java/io/scalecube/account/api/OrganizationMember.java]:
```suggestion
        .add("id='").add(id).add("'")
```

File: scalecube-organization-api/src/main/java/io/scalecube/account/api/OrganizationMember.java [hunk 2]
```diff
@@ -45,6 +46,9 @@ public int hashCode() {
...
+        .add("role='" + role + "'")
```
Comment (aharonha) [inline scalecube-organization-api/src/main/java/io/scalecube/account/api/OrganizationMember.java]:
```suggestion
        .add("role='").add(role).add("'")
```

File: scalecube-organization/src/main/java/io/scalecube/organization/domain/Organization.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+import io.scalecube.account.api.Role;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.StringJoiner;
+
+/** Represents an Organization. */
+public class Organization extends Entity {
+
+  private String id;
```
Comment (segabriel) [inline scalecube-organization/src/main/java/io/scalecube/organization/domain/Organization.java]: you also have such a field in super class

File: scalecube-organization/src/main/java/io/scalecube/organization/operation/CreateOrganization.java [hunk 4]
```diff
@@ -47,31 +44,29 @@ protected CreateOrganizationResponse process(
...
-                .name(request.name())
-                .email(request.email())
-                .keyId(UUID.randomUUID().toString())
-                .build());
+      CreateOrganizationRequest request, OperationServiceContext context, String id) {
+    Organization organization =
+        new Organization(
+            id,
+            request.name(),
+            request.email(),
```
Comment (aharonha) [inline scalecube-organization/src/main/java/io/scalecube/organization/operation/CreateOrganization.java]: `context.profile()` is removed. what was its purpose and why we should not use it from now on?

File: scalecube-organization/src/main/java/io/scalecube/organization/operation/GetOrganizationMembers.java [hunk 5]
```diff
@@ -21,11 +21,9 @@ protected GetOrganizationMembersResponse process(
     Organization organization = getOrganization(request.organizationId());
 
     checkSuperUserAccess(organization, context.profile());
-    Collection<OrganizationMember> organizationMembers =
-        context.repository().getOrganizationMembers(organization);
-    OrganizationMember[] members = new OrganizationMember[organizationMembers.size()];
 
-    return new GetOrganizationMembersResponse(organizationMembers.toArray(members));
+    return new GetOrganizationMembersResponse(
+        organization.members().toArray(new OrganizationMember[0]));
```
Comment (aharonha) [inline scalecube-organization/src/main/java/io/scalecube/organization/operation/GetOrganizationMembers.java]: why did you set the array size to `0`?

File: scalecube-organization/src/main/java/io/scalecube/organization/operation/GetUserOrganizationsMembership.java [hunk 6]
```diff
@@ -5,29 +5,28 @@
...
-            .repository()
-            .getUserMembership(context.profile().getUserId())
-            .stream()
+        StreamSupport.stream(context.repository().findAll().spliterator(), false)
+            .filter(organization -> organization.isMember(context.profile().getUserId()))
             .map(
-                item -> {
-                  Role role = getRole(context.profile().getUserId(), item);
-                  return organizationInfo(item, apiKeyFilterBy(role)).build();
+                organization -> {
```
Comment (aharonha) [inline scalecube-organization/src/main/java/io/scalecube/organization/operation/GetUserOrganizationsMembership.java]: :+1:

File: scalecube-organization/src/main/java/io/scalecube/organization/operation/OrganizationInfoOperation.java [hunk 7]
```diff
@@ -26,7 +25,7 @@ protected void validate(OrganizationInfo info, OperationServiceContext context)
               + "underscore, period, dash & percent");
     }
 
-    if (context.repository().existByName(info.name())) {
+    if (context.repository().existsByName(info.name())) {
```
Comment (aharonha) [inline scalecube-organization/src/main/java/io/scalecube/organization/operation/OrganizationInfoOperation.java]: :+1:

File: scalecube-organization/src/main/java/io/scalecube/organization/operation/ServiceOperation.java [hunk 8]
```diff
@@ -96,11 +90,10 @@ protected GetOrganizationResponse getOrganizationResponse(
 
   protected OrganizationInfo.Builder organizationInfo(
       Organization organization, Predicate<ApiKey> filter) {
-    ApiKey[] apiKeys = Arrays.stream(organization.apiKeys()).filter(filter).toArray(ApiKey[]::new);
     return OrganizationInfo.builder()
         .id(organization.id())
         .name(organization.name())
-        .apiKeys(apiKeys)
+        .apiKeys(organization.apiKeys().stream().filter(filter).toArray(ApiKey[]::new))
```
Comment (aharonha) [inline scalecube-organization/src/main/java/io/scalecube/organization/operation/ServiceOperation.java]: :+1: that's a better option to make an array, but why do we use arrays and not `List`s?

File: scalecube-organization/src/main/java/io/scalecube/organization/repository/couchbase/CouchbaseRepository.java [hunk 9]
```diff
@@ -1,183 +1,169 @@
...
+import java.util.concurrent.TimeoutException;
+import rx.Observable;
+
+/**
+ * Abstract base couchbase <Code>Repository</Code> implementation.
+ *
+ * @param <T> This repository entity type.
+ * @param <I> This repository entity Id type which extends <codes>java.lang.String</codes>.
+ */
+abstract class CouchbaseRepository<T extends Entity, I extends String> implements Repository<T, I> {
```
Comment (aharonha) [inline scalecube-organization/src/main/java/io/scalecube/organization/repository/couchbase/CouchbaseRepository.java]: `String` is `final` in java. Can we just remove `I`?

File: scalecube-organization/src/test/java/io/scalecube/organization/Base.java [hunk 10]
```diff
@@ -102,21 +94,17 @@
...
     organizationRepository = new InMemoryOrganizationRepository();
-    admin = new InMemoryOrganizationMembersRepositoryAdmin();
     service = createService(testProfile);
     new File("keystore.properties").deleteOnExit();
   }
 
   @BeforeAll
   static void beforeAll() throws NoSuchAlgorithmException {
-    keyPairGenerator = KeyPairGenerator.getInstance("RSA");
+    KeyPairGenerator keyPairGenerator = KeyPairGenerator.getInstance("RSA");
```
Comment (aharonha) [inline scalecube-organization/src/test/java/io/scalecube/organization/Base.java]: why did you remove this static member?

File: scalecube-organization/src/main/java/io/scalecube/organization/OrganizationServiceImpl.java [hunk 11]
```diff
@@ -92,7 +92,7 @@ public OrganizationServiceImpl(
             logger.debug("createOrganization: exit, return: {}", response);
             result.success(response);
           } catch (ServiceOperationException ex) {
-            logger.error("createOrganization: ERROR: {}", ex);
+            logger.error("createOrganization: ERROR", ex);
```
[the remark this answers is not in the document]
Comment (dmytro-lazebnyi) [reply scalecube-organization/src/main/java/io/scalecube/organization/OrganizationServiceImpl.java]: To see the whole stack trace.

File: scalecube-organization/src/main/java/io/scalecube/organization/OrganizationServiceImpl.java [hunk 12]
```diff
@@ -117,7 +117,7 @@ public OrganizationServiceImpl(
                 response.organizations().length);
             result.success(response);
           } catch (ServiceOperationException ex) {
-            logger.error("getUserOrganizationsMembership: ERROR: {}", ex);
+            logger.error("getUserOrganizationsMembership: ERROR", ex);
```
[the remark this answers is not in the document]
Comment (dmytro-lazebnyi) [reply scalecube-organization/src/main/java/io/scalecube/organization/OrganizationServiceImpl.java]: To see the whole stack trace.

File: scalecube-organization/src/main/java/io/scalecube/organization/domain/Organization.java [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+    this.email = requireNonNull(email, "organization email cannot be null");
+    this.keyId = requireNonNull(keyId, "organization keyId cannot be null");
+
+    addMember(
+        new OrganizationMember(
+            requireNonNull(creatorUserId, "organization creator id cannot be null"),
+            Role.Owner.name()));
+  }
+
+  public String id() {
```
[the remark this answers is not in the document]
Comment (dmytro-lazebnyi) [reply scalecube-organization/src/main/java/io/scalecube/organization/domain/Organization.java]: This `getter` is already removed from `Organization`.

File: scalecube-organization/src/main/java/io/scalecube/organization/domain/Organization.java [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+   * Creates new instance of organization.
+   *
+   * @param id organization id.
+   * @param name organization name.
+   * @param email organization email.
+   * @param keyId organization key id.
+   * @param creatorUserId user id of organization creator.
+   */
+  public Organization(String id, String name, String email, String keyId, String creatorUserId) {
+    this.id = requireNonNull(id, "organization id cannot be null");
```
[the remark this answers is not in the document]
Comment (dmytro-lazebnyi) [reply scalecube-organization/src/main/java/io/scalecube/organization/domain/Organization.java]: No redundant builders anymore. All passed arguments are mandatory. 
Exception in the constructor is a good way to indicate a problem with object creation.

File: scalecube-organization/src/main/java/io/scalecube/organization/operation/CreateOrganization.java (hunk 4 above)
Comment (dmytro-lazebnyi) [reply scalecube-organization/src/main/java/io/scalecube/organization/operation/CreateOrganization.java]: Previously it was used only for retrieving userId of the creator in the `OrganizationsDataAccessImpl`.
Now it is redundant since userId is retrieved and set to `Organization` object right here (in the `CreateOrganization`).

File: scalecube-organization/src/main/java/io/scalecube/organization/operation/GetOrganizationMembers.java (hunk 5 above)
Comment (dmytro-lazebnyi) [reply scalecube-organization/src/main/java/io/scalecube/organization/operation/GetOrganizationMembers.java]: https://shipilev.net/blog/2016/arrays-wisdom-ancients/

File: scalecube-organization/src/main/java/io/scalecube/organization/operation/ServiceOperation.java (hunk 8 above)
Comment (dmytro-lazebnyi) [reply scalecube-organization/src/main/java/io/scalecube/organization/operation/ServiceOperation.java]: I don't know why it (arrays in `OrganizationInfo) was designed in this way.
But I didn't change it since I avoided changing the public API in this PR.

File: scalecube-organization/src/main/java/io/scalecube/organization/repository/couchbase/CouchbaseRepository.java (hunk 9 above)
Comment (dmytro-lazebnyi) [reply scalecube-organization/src/main/java/io/scalecube/organization/repository/couchbase/CouchbaseRepository.java]: Done.

File: scalecube-organization/src/test/java/io/scalecube/organization/Base.java (hunk 10 above)
Comment (dmytro-lazebnyi) [reply scalecube-organization/src/test/java/io/scalecube/organization/Base.java]: Removed it at all. It is not used in these tests.
